gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/analytics_engine/permissions/ApiAnalyticsQueryFilterDecoratorImpl.java

```diff
@@ -44,44 +56,136 @@
+        var updatedFilters = updateContextFilters(context.filters(), allowedApis);
+
+        return context.withApiNamesById(allowedApis).withApplicationNameById(allowedApplications).withFilters(updatedFilters);
+    }
+
+    private Map<String, String> getAllowedApis(List<Filter> filters) {
+        QueryBuilder<ApiEntity> queryBuilder = QueryBuilder.create(ApiEntity.class);
+
+        var apiIds = getIdsFromFilters(filters, API);
+        if (!(apiIds.isEmpty())) {
+            queryBuilder.setFilters(Map.of("id", apiIds));
+        }
+
+        boolean admin = isAdmin();
+        Principal userPrincipal = getUserPrincipal();
+        String name = userPrincipal.getName();
+        ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+        Page<GenericApiEntity> search = apiSearchService.search(
+            executionContext,
+            name,
+            admin,
+            queryBuilder,
+            new PageableImpl(0, Integer.MAX_VALUE),
+            false,
+            true
+        );
+
+        List<GenericApiEntity> content = search.getContent();
+
+        return mapApiIdsToNames(content);
+    }
+
+    private Map<String, String> getAllowedApplications(List<Filter> filters) {
+        var query = new ApplicationQuery();
+
+        var applicationIds = getIdsFromFilters(filters, APPLICATION);
+        if (!(applicationIds.isEmpty())) {
+            query.setIds(applicationIds);
+        }
+
+        if (!isAdmin()) {
+            query.setUser(getAuthenticatedUser());
+        }
+
+        List<ApplicationListItem> content = applicationSearchService
+            .search(GraviteeContext.getExecutionContext(), query, null, null)
+            .getContent();
+        return mapApplicationIdsToNames(content);
+    }
+
+    /**
+     * getIdsFromFilters returns all the IDs found in a list of filters.
+     *
+     * @param filterName specifies the filters to look for
+     */
+    public static Set<String> getIdsFromFilters(List<Filter> filters, FilterSpec.Name filterName) {
+        if (filters.isEmpty()) {
+            return Set.of();
         }
 
-        apiAuthorizationService
-            .findIdsByUser(executionContext, getAuthenticatedUser(), false)
+        List<Filter> filtersByName = filters
             .stream()
-            .filter(appId -> permissionService.hasPermission(executionContext, API_ANALYTICS, appId, READ))
-            .forEach(id -> allowedApis.put(id, new API("")));
+            .filter(filter -> filter.name() == filterName)
+            .toList();
 
-        return allowedApis;
+        if (filtersByName.isEmpty()) {
+            return Set.of();
+        }
+
+        List eqFilters = filtersByName
+            .stream()
+            .filter(filter -> filter.operator() == EQ)
+            .map(Filter::value)
+            .toList();
+
+        var objectMapper = new ObjectMapper();
+        List inFilters = filtersByName
```

Comment:
Do we really need to rely on an object mapper here ? This make me even more willing not to do this additional processing of input filters ...